src/content/components/TreeView.js

```diff
@@ -70,11 +76,20 @@ class TreeViewRowFixedColumns extends Component {
     return (
       <div className={`treeViewRow treeViewRowFixedColumns ${evenOddClassName} ${selected ? 'selected' : ''}`} style={{height: '16px'}} onMouseDown={this._onClick}>
         {
-          columns.map(col =>
-            <span className={`treeViewRowColumn treeViewFixedColumn ${col.propName}`}
-                  key={col.propName}>
-              { reactStringWithHighlightedSubstrings(node[col.propName], highlightString, 'treeViewHighlighting') }
-            </span>)
+          columns.map(col => {
+            let renderedComponent;
+            if (col.propName === 'icon') {
+              renderedComponent = <NodeIcon node={ node } onDisplayIcon={ icon => this.props.onDisplayIcon(icon) }/>;
```

Comment:
I'm not really happy with this but don't have a better solution :/ In a previous version of the patch I passed "NodeIcon" as a prop to TreeView from ProfileTreeView but now that I want to both handle onDisplayIcon and pass a node I don't think it's possible -- or easy.